examples/nim/default.nim

```diff
@@ -0,0 +1,9 @@
+# Minimal example
+# Allowed commands:
+#    - "compile", "compileToC", "c" to build C assambly
+#    - "compileToCpp", "cpp", "cc" to build C++ assambly
+#    - "compileToOC", "objc" to build objective-C assambly
+#    - "js" to output js code
+#    - "check" to check code validity
+
+echo "hello world"
```

Comment:
Default examples for the rest of the languages are a function that returns the square of the given number. Is there any specific reason for ir not to be the same for nim?

Reply:
no that's a mistake on my part.
Trying to replace it with the default example highlights an issue though.
The default example will trigger a hint 'square defined but not used'.
This is because Nim considers by default that the file is a program and not a module.
To fix this there is the --genScript option that create an sh script that builds the object file in the nimcache folder but stops short of calling it.
Because of this if we want to use this option we then have to invoke it ourselves by navigating inside the cache folder.
Nim really isn't making it easy to simply built an object file :cry: 

What's your thought on this?

I've seen that an --asm option was added but couldn't get a version of nim that add it.

Reply:
You can use 

```nim
proc sqrt() =
   ## function logic

when isMainModule():
    sqrt()
```

Reply:
> Nim really isn't making it easy to simply built an object file 

Use an explicit ``--nimcache`` setting?

Reply:
<img width="807" alt="Capture d’écran 2019-12-26 à 20 30 08" src="https://user-images.githubusercontent.com/21139416/71488044-c35cf300-281e-11ea-9e9b-fc0806051681.png">

Interesting settings for the work

Reply:
> > Nim really isn't making it easy to simply built an object file
> 
> Use an explicit `--nimcache` setting?

It's already what I'm doing

Reply:
> no that's a mistake on my part.
> Trying to replace it with the default example highlights an issue though.
> The default example will trigger a hint 'square defined but not used'.
> This is because Nim considers by default that the file is a program and not a module.
> To fix this there is the --genScript option that create an sh script that builds the object file in the nimcache folder but stops short of calling it.
> Because of this if we want to use this option we then have to invoke it ourselves by navigating inside the cache folder.
> Nim really isn't making it easy to simply built an object file 
> 
> What's your thought on this?
> 
> I've seen that an --asm option was added but couldn't get a version of nim that had it.

I seem to have overthought that, I'm adding the new example